### DISCUSSION TITLE
Getting type error at [...nextauth]/route.ts

### DISCUSSION THREAD

**Discussion by vaibhav135** [nextauthjs/next-auth#8780] [Help]
### Summary

I am implementing credentials (database sessions) for my auth and followed this code from this github issue -> https://github.com/nextauthjs/next-auth/discussions/4394#discussioncomment-5503602. @decovicdev is this working fine for you.

next-auth version: 4.23.2

### Additional information

**Here's the error that I am getting.**

```bash

.next/types/app/api/auth/[...nextauth]/route.ts:8:13 - error TS2344: Type 'OmitWithTag<typeof import("/Users/someuser/project/src/app/api/auth/[...nextauth]/route"), "POST" | "config" | "generateStaticParams" | "revalidate" | ... 11 more ... | "PATCH", "">' does not satisfy the constraint '{ [x: string]: never; }'.
  Property 'authOptionsWrapper' is incompatible with index signature.
 Type '(request: NextRequest, context: Context) => [NextRequest, Context, AuthOptions]' is not assignable to type 'never'.

  8 checkFields<Diff<{
                ~~~~~~
  9   GET?: Function
    ~~~~~~~~~~~~~~~~
...
 25

 26 }, TEntry, ''>>()
    ~~~~~~~~~~~~~~


Found 1 error in .next/types/app/api/auth/[...nextauth]/route.ts:8

error Command failed with exit code 2.
```

**Comment by solo-samurai** [comment]
Hello @vaibhav135, please check out https://github.com/decovicdev/next-auth-database-session/tree/main/apps/app
this app uses `next: 13.5.3`, and `next-auth: 4.23.1`, and I had no type error.  I'm working on fixing `turborepo` errors and publishing live site.

[ Github Repo Link](https://github.com/decovicdev/next-auth-database-session)

[Live Site Link](https://next-auth-database-session-app.vercel.app/)

**Reply by vaibhav135** [reply]
This is very weird. Once I am deleting the `node_module` and `.next` folder and do `npm i`, the error disappears. And pops up again after some time.

**Reply by solo-samurai** [reply]
There's something somewhere in the code, I haven't come across that error. you can share your repo if you can and I will try to debug with you.

**Reply by vaibhav135** [reply]
Sure here's my repo: https://github.com/vaibhav135/url-shortner/

**Reply by solo-samurai** [reply]
@vaibhav135  I looked at the code, and I couldn't run it because of small errors like, context provider components should be client, and empty middleware shouldn't be created once I fixed I could run it, also make sure you await promises on signup wrote
```ts
  const newUser = await prisma.user.create({
    data: {
      email,
      name: user,
      password: hashedPassword,
    },
  })

  prisma.account.create({
    data: {
      userId: newUser.id,
      type: 'credentials',
      provider: 'credentials',
      providerAccountId: newUser.id,
    },
  })
  ```

**Reply by vaibhav135** [reply]
Aaah yes thanks for pointing it out. Actually it's still wip. And I am adding all these things in the next commit.

**Reply by solo-samurai** [reply]
Oh Ok, upgrade to the current version of both `next` and `next auth`, look at the example repo for reference, and when you build your auth flow if you still have issues let me know. all the best.

**Reply by solo-samurai** [reply]
@vaibhav135 example repo is live now. if this is useful please make my comment answer for the discussion. thanks and all the best.

**Reply by vaibhav135** [reply]
Thanks
